estuary/utils/general.py

```diff
@@ -366,3 +352,32 @@ def format_story_results(results, requested_item):
             'requested_node_index': requested_node_index
         }
     }
+
+
+def organize_labels(requested_node_label, results, reverse=False):
+    """
+    Replace Neo4j labels with appropriate labels of the story flow.
+
+    :param string requested_node_label: label of the node requested by the user
+    :param list results: nodes in a story/path
+    :kwarg bool reverse: determines if the results are in reverse order of the story flow
+    :return: results with story/path labels
+    :rtype: list
+    """
+    # Avoid circular imports
+    from estuary.models import story_flow_list
+
+    if results:
+        idx = 0
+        if reverse:
+            idx = len(results) - 1
+        node_idx = story_flow_list.index(requested_node_label)
+        while (0 <= idx < len(results)):
```

Comment:
Another way to do this might be to say:

```python
# Increment or decrement through the list, depending.
delta = -1 if reverse else 1
while (0 <= idx < len(results)):
    results[idx].__label__ = story_flow_list[node_idx]
    idx += delta
    node_idx += delta
```

Reply:
Yeah, I like this change